src/libostree/ostree-repo-refs.c

```diff
@@ -535,21 +540,32 @@ ostree_repo_resolve_collection_ref (OstreeRepo                    *self,
 
   /* Check for the ref in the current transaction in case it hasn't been
    * written to disk, to match the behavior of ostree_repo_resolve_rev() */
-  if (self->in_transaction)
+  if (self->in_transaction && self->txn.collection_refs)
```

Comment:
I don’t think you can access `self->txn.collection_refs` without holding the `txn_lock`, since it could be initialised at any time by another thread calling (for example) `ostree_repo_transaction_set_refspec()` (i.e. `ensure_txn_refs()`).

Reply:
Good point. In that case I opened https://github.com/ostreedev/ostree/issues/1826 because that rule doesn't seem to be followed consistently